### ISSUE TITLE
CenterOnScreen not working for FixedSize Window

### ISSUE THREAD

**Issue by scorpione71** [fyne-io/fyne#2550]
### Describe the bug:
 Good evening. I have some problems found after updating Fyne V. 2.0 to Version V. 2.1. The fields of the struct Label "Alignment fyne.TextAlign" and "TextStyle fyne.TextStyle", which worked perfectly with V. 2.0, do not work after the update to V. 2.1. Another problem, the "CenterOnScreen ()" function, which should place the window in the center of the screen, which worked before the update, with the new version, unfortunately it positions the window lower and to the right of the center of the screen. I hope that these problems that I have encountered can be solved, and in any case, I take advantage of the opportunity to compliment the excellent work done.

### To Reproduce:
Steps to reproduce the behaviour:

### Screenshots:
 
![Allign Style](https://user-images.githubusercontent.com/58878392/137344393-7e80cbfb-6766-4c03-b539-79612a86b470.jpg)

![CenterOnScreen](https://user-images.githubusercontent.com/58878392/137344411-c0721087-4823-418f-9a33-6e621de39d7d.jpg)
 

### Device (please complete the following information):
 - **OS:** Linux Mint 
 - **Version:**  20.2] 
 - **Go version:** 1.17.2]
 - **Fyne version:** 2.1]

**Comment by scorpione71** [comment]
Thank you for answering. It was very useful. If you find a solution for the
CenterOnScreen() function as well, please let me know. Thank you very much
for the attenzione.

Il ven 15 ott 2021, 15:16 Andy Williams ***@***.***> ha
scritto:

**Comment by andydotxyz** [comment]
I cannot replicate the center on screen issue, perhaps it is Linux (Mint?) specific.
Can you provide more details about the desktop / window manager so we can replicate?

**Comment by lucasepe** [comment]
Maybe the center on screen is not a real issue.
If it can help, I noticed that this also happens to me but only when I activate the second monitor. 
Using just one display the window is correctly centered on the screen. 

```sh
$ hostnamectl
  Operating System: Debian GNU/Linux 11 (bullseye)
            Kernel: Linux 5.10.0-9-amd64
      Architecture: x86-64
```

```sh
$ cat /usr/share/gnome/gnome-version.xml
<?xml version="1.0"?>
<gnome-version>
  <platform>3</platform>
  <minor>38</minor>
  <micro>5</micro>
  <distributor>Debian</distributor>
</gnome-version>
```

**Comment by andydotxyz** [comment]
> Using just one display the window is correctly centered on the screen.

When there is a second screen active where does the window appear?

**Comment by andydotxyz** [comment]
That could be helpful thanks. There a couple of different ways the positioning could be off and seeing it will help.

**Comment by lucasepe** [comment]
I have to correct myself, doing more experiments I noticed that regardless of number of monitors the window is not centered when the `w.Resize(...)` command is given (calls order does not make difference).

I have attached an animated gif showing that using `w.Resize(...)` the window is no longer centered (btw: I'dont think it's a big issue...users move windows all the time).
![center_on_screen](https://user-images.githubusercontent.com/585381/141956260-0eddbd7f-382f-40b4-8088-e310bd01c3bb.gif)

**Comment by andydotxyz** [comment]
Fixed in v2.1.2

**Comment by scorpione71** [comment]
Good evening, thanks for the information. I'll keep practicing with Fyne. I
think he has already reached a good maturity, I hope that many more tools
will be introduced in the future.
Good job.

Il giorno mar 7 dic 2021 alle ore 01:12 Andy Williams <
***@***.***> ha scritto:
